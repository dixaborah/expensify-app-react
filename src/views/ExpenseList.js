import React from 'react'
import { connect } from 'react-redux'
import ExpenseListItem from '../views/ExpenseListItem'
import selectExpenses from '../selectors/expenses'

// Stateless Component
export const ExpenseList = (props) => (
    <div>
    {
        props.expenses.length === 0 ? (
          <p>No Expense found</p>
        ) : (
          props.expenses.map((expense) => {
            return <ExpenseListItem key={expense.id} {...expense} />;
          })
        )
      }
    </div>
)

const mapStateToProps = (state) => ({
    expenses: selectExpenses(state.expenses, state.filters)
})

export default connect(mapStateToProps)(ExpenseList)