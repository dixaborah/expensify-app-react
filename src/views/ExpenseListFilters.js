import React from 'react'
import { connect } from 'react-redux'
import { DateRangePicker } from 'react-dates'
import { setEndDateFilter, setStartDateFilter, setTextFilter, sortByAmountFilter, sortByDateFilter } from '../actions/filters'


export class ExpenseListFilters extends React.Component {
    state = {
        calenderFocused: null
    }

    onDatesChange = ({ startDate, endDate }) => {
        this.props.setStartDateFilter(startDate)
        this.props.setEndDateFilter(endDate)
    }

    onFocusChange = (calenderFocused) => {
        this.setState(() => ({ calenderFocused }))
    }

    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value)
    }

    onSelectChange = (e) => {
        e.target.value.includes('date') ? 
        this.props.sortByDateFilter({date: e.target.value}) : 
        this.props.sortByAmountFilter({amount: e.target.value})
    }

    render () {
        return (
            <div>
                <input type="text" value={this.props.filters.text} onChange={this.onTextChange} />

                <select value={this.props.filters.sortBy} onChange={this.onSelectChange}>
                    <option value="date_recent">Newest</option>
                    <option value="date_oldest">Oldest</option>
                    <option value="amount_lowest">Lowest Price</option>
                    <option value="amount_highest">Highest Price</option>
                </select>

                <DateRangePicker
                    startDate={this.props.filters.startDate}
                    endDate={this.props.filters.endDate}
                    onDatesChange={this.onDatesChange}
                    focusedInput={this.state.calenderFocused}
                    onFocusChange={this.onFocusChange}
                    numberOfMonths={1}
                    isOutsideRange={() => false}
                    showClearDates={true}
                />
            </div>
        )
    }
}


const mapStateToProps = (state) => ({ filters: state.filters })

const mapDispatchToProps = (dispatch) => ({
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    sortByAmountFilter: (amount) => dispatch(sortByAmountFilter(amount)),
    sortByDateFilter: (date) => dispatch(sortByDateFilter(date)),
    setStartDateFilter: (startDate) => dispatch(setStartDateFilter(startDate)),
    setEndDateFilter: (endDate) => dispatch(setEndDateFilter(endDate))

})

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters)