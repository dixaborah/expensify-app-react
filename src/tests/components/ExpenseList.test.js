import React from 'react'
import { shallow } from 'enzyme'

import { ExpenseList } from '../../views/ExpenseList'
import Expenses from '../fixtures/Expenses'



test('Should render ExpenseList with Expenses', () => {
    const wrapper = shallow(<ExpenseList expenses={Expenses} />)
    expect(wrapper).toMatchSnapshot()
})

test('Should render ExpenseList with empty message', () => {
    const wrapper = shallow(<ExpenseList expenses={[]} />)
    expect(wrapper).toMatchSnapshot()
})