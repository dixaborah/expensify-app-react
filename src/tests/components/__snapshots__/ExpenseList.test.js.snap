// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpenseList with Expenses 1`] = `
<div>
  <ExpenseListItem
    amount={195}
    createdAt={1420070400000}
    description="Gum"
    id={1}
    key="1"
    note=""
  />
  <ExpenseListItem
    amount={109500}
    createdAt={1483228800000}
    description="Rent"
    id={2}
    key="2"
    note=""
  />
  <ExpenseListItem
    amount={95075900}
    createdAt={1546300800000}
    description="Lamboghini"
    id={3}
    key="3"
    note=""
  />
</div>
`;

exports[`Should render ExpenseList with empty message 1`] = `
<div>
  <p>
    No Expense found
  </p>
</div>
`;
